### PR TITLE
Preview Breaking changes: RecoveryServices Azure Site Recovery : API version bumpup 

### PR DESCRIPTION
**Pull request by viverm**
## New feature support for Azure Site Recovery.
Reference::
https://github.com/Azure/azure-powershell/pull/9173

Preview Changes 
Cmdlet Review :: 
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/300#issuecomment-491092895

**Changes** 
* Azure Site Recovery Support to write to managed disk for vMWare to Azure. 
* Azure Site Recovery Support to NIC reduction for vMWare to Azure. 
* Azure Site Recovery Support to accelerated networking for Azure to Azure. 
* Azure Site Recovery Support to agent auto update for Azure to Azure. 
* Azure Site Recovery Support to Standard SSD for Azure to Azure. 
* Azure Site Recovery Support to Azure Disk Encryption two pass for Azure to Azure. 
* Azure Site Recovery Support to protect newly added disk for Azure to Azure.

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by viverm** [comment]
@cormacpayne  / @MiYanni  Build is failing in running test cases for Other provider.
What can we do to fix it?

**Comment by MiYanni** [comment]
@viverm Update your branch with latest from Az.RecoveryServices once this PR is merged: https://github.com/Azure/azure-powershell/pull/9352

**Comment by viverm** [comment]
image
image

File: src/RecoveryServices/RecoveryServices/help/Backup-AzRecoveryServicesBackupItem.md [hunk 1]
```diff
@@ -58,11 +58,26 @@ Accept pipeline input: False
...
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
 ### -DefaultProfile
 The credentials, account, tenant, and subscription used for communication with azure.
 
 ```yaml
-Type: Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer
+Type: IAzureContextContainer
```
Comment (MiYanni) [inline src/RecoveryServices/RecoveryServices/help/Backup-AzRecoveryServicesBackupItem.md]: You are missing the `-UseFullTypeName` parameter when regenerating documentation.
https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module

Comment (viverm) [reply src/RecoveryServices/RecoveryServices/help/Backup-AzRecoveryServicesBackupItem.md]: updated.

File: src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/NewAzureRmRecoveryServicesAsrReplicationProtectedItem.cs [hunk 2]
```diff
@@ -729,9 +783,49 @@ private void AzureToAzureReplication(EnableProtectionInput input)
...
             input.Properties.ProviderSpecificDetails = providerSettings;
         }
 
+        private DiskEncryptionInfo A2AEncryptionDetailDetails()
+        {
+            // Any encryption data is present.
+            if (this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.DiskEncryptionSecretUrl)) ||
+                this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.DiskEncryptionVaultId)) ||
+                this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.KeyEncryptionKeyUrl)) ||
+                this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.KeyEncryptionVaultId)))
```
Comment (MiYanni) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/NewAzureRmRecoveryServicesAsrReplicationProtectedItem.cs]: You can just use `this.IsParameterBound(c => c.DiskEncryptionSecretUrl)`, etc. You used it about 300 lines up, but didn't use it for any of the logic below.

File: src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/UpdateAzureRmRecoveryServicesAsrProtectionDirection.cs [hunk 3]
```diff
@@ -716,6 +769,45 @@ private void validateRPISwitchParam()
             }
         }
 
+        private DiskEncryptionInfo A2AEncryptionDetails()
+        {
+            // Any encryption data is present.
+            if (this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.DiskEncryptionSecretUrl)) ||
+                this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.DiskEncryptionVaultId)) ||
+                this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.KeyEncryptionKeyUrl)) ||
+                this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.KeyEncryptionVaultId)))
```
Comment (MiYanni) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/UpdateAzureRmRecoveryServicesAsrProtectionDirection.cs]: Same here with `this.IsParameterBound` usage.

**Comment by MiYanni** [comment]
@viverm I didn't know if you wanted to change the above recommendations before I merge the PR.

**Pull request merged**